samples/SampleApp/PooledHttpContextFactory.cs

```diff
@@ -1,24 +1,56 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using System.Text;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Http.Features.Internal;
+using Microsoft.Extensions.ObjectPool;
 
 namespace SampleApp
 {
     public class PooledHttpContextFactory : IHttpContextFactory
     {
+        private readonly ObjectPool<StringBuilder> _builderPool;
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly Stack<PooledHttpContext> _pool = new Stack<PooledHttpContext>();
 
+        public PooledHttpContextFactory()
+            : this(poolProvider: null, httpContextAccessor: null)
+        {
+        }
+
+        public PooledHttpContextFactory(ObjectPoolProvider poolProvider)
+            : this(poolProvider, httpContextAccessor: null)
+        {
+        }
+
         public PooledHttpContextFactory(IHttpContextAccessor httpContextAccessor)
```

Comment:
Having this be the only constructor would mean the class could only be instantiated if an `IHttpContextAccessor` were registered. sad :panda_face: